docs/api-reference.md

```diff
@@ -0,0 +1,273 @@
+and the second index represents `x` columns.
+If `x` is a pandas `DataFrame` or `Series`, the result is a `DataFrame`
+with column names and lag values as the row indices.
+
+The time lag $\Delta$ is interpreted as $y(t) \sim x(t - \Delta) | z(t - \Delta_{\mathrm{cond}})$.
+The time lags are applied to the `x` and `cond` arrays such that the `y`
+array stays the same every time.
+This means that `y` is cropped to `y[max(max_lag,0) : N+min(min_lag,0)]`.
+The `cond_lag` parameter specifies the lag for the `cond` array separately
+from the `x` lag.
+
+If the `mask` parameter is set, only those `y` observations with the
+matching mask element set to `True` are used for estimation.
+
+## Positional or keyword parameters:
+- `y: array_like`.
+  
+  A 1D array of observations. If `discrete_y` is True, the values may be
+  of any type. Otherwise the values must be numeric.
+
+- `x: array_like`.
+
+  A 1D or 2D array where the columns are one or more variables and the
+  rows are observations. The number of rows must be the same as in y.
+
+- `lag: array_like`, default 0.
+  
+  A time lag or 1D array of time lags to apply.
+  A positive lag means that `y` depends on earlier `x` values and vice versa.
+
+## Optional keyword parameters:
+- `k: int`, default 3.
+  
+  The number of neighbors to consider.
+
+- `cond: array_like`, default `None`.
+  
+  Optional 1D or 2D array of observations used for conditioning.
+  Must have as many observations as `y`.
+  A $(n \times m)$ array is interpreted as a single $m$-dimensional variable.
+
+- `cond_lag: array_like`, default 0.
+
+  Lag applied to the cond array. Must be broadcastable to the size of `lag`.
+  Can be two-dimensional to lag each conditioning variable separately.
+
+- `mask: array_like` or `None`.
+
+  If specified, an array of booleans that gives the `y` elements to use for
+  estimation. Use this to exclude some observations from consideration
+  while preserving the time series structure of the data. Elements of
+  `x` and `cond` are masked with the lags applied.
+
+- `discrete_y: bool`, default False.
+
+  If True, the `y` variable is interpreted as a discrete variable. The `x`
+  variables are still continuous. The `y` values may be non-numeric.
+
+- `preprocess: bool`, default True.
+
+  By default, the variables are scaled to unit variance and
+  added with low-amplitude noise. The noise uses a fixed random seed.
+
+- `drop_nan: bool`, default False.
+
+  If True, all NaN (not a number) values in `x`, `y` and `cond` are masked out.
+
+- `normalize: bool`, default False.
+
+  If True, the results will be normalized to correlation coefficient scale.
+  Same as calling `normalize_mi` on the results.
+
+- `max_threads: int` or `None`.
+  
+  The maximum number of threads to use for estimation.
+  By default, the number of CPU cores is used.
+  Because the calculation is CPU-bound, more threads should not be used.
+
+- `callback: method` or `None`.
```

Comment:
Mention the example code snippet for using this. The snippet could include a usage example.